src/apps/filters/h3.c

```diff
@@ -1189,6 +1189,433 @@ SUBCOMMAND(uncompactCells,
+                        polygonString[i - strPos] = polygonString[i];
+                    }
+                    // Set the string position to the end of the buffer
+                    strPos = 1500 - strPos;
+                    // Grab up to the remaining size of the buffer and fill it
+                    // into the file
+                    // Did you know that if the amount you want to read from
+                    // the file buffer is larger than the buffer itself,
+                    // fread can choose to give you squat and not the
+                    // remainder of the file as you'd expect from the
+                    // documentation? This was a surprise to me and a
+                    // significant amount of wasted time to figure out how
+                    // to tackle. C leaves *way* too much as undefined
+                    // behavior to be nice to work with...
+                    int64_t i = 0;
+                    int res;
+                    do {
+                        res = fread(polygonString + strPos + i, 1, 1, fp);
+                        i++;
+                    } while (i < 1500 - strPos && res != 0);
+                    if (res == 0) {
+                        polygonString[strPos + i - 1] = 0;
+                    }
+                    strPos = 0;
+                }
+            } else {
+                strPos++;
+            }
+        }
+    }
+    free(verts);
+    return E_SUCCESS;
+}
+
+SUBCOMMAND(
+    polygonToCells,
+    "Converts a polygon (array of lat, lng points, or array of arrays of lat, "
+    "lng points) into a set of covering cells at the specified resolution") {
+    char filename[1024] = {0};  // More than Windows, lol
+    Arg filenameArg = {
+        .names = {"-f", "--file"},
+        .scanFormat = "%1023c",
+        .valueName = "FILENAME",
+        .value = &filename,
+        .helpText =
+            "The file to load the polygon from. Use -- to read from stdin."};
+    char polygonStr[1501] = {0};  // Up to 100 cells with zero padding
+    Arg polygonStrArg = {
+        .names = {"-p", "--polygon"},
+        .scanFormat = "%1500c",
+        .valueName = "POLYGON",
+        .value = &polygonStr,
+        .helpText = "The polygon to convert. Up to 1500 characters."};
+    int res = 0;
+    Arg resArg = {.names = {"-r", "--resolution"},
+                  .required = true,
+                  .scanFormat = "%d",
+                  .valueName = "res",
+                  .value = &res,
+                  .helpText =
+                      "Resolution, 0-15 inclusive, that the compacted set "
+                      "should be uncompacted to. Must be greater than or equal "
+                      "to the highest resolution within the compacted set."};
+    Arg *args[] = {&polygonToCellsArg, &helpArg, &filenameArg, &polygonStrArg,
+                   &resArg};
+    PARSE_SUBCOMMAND(argc, argv, args);
+    if (!filenameArg.found && !polygonStrArg.found) {
```

Comment:
```suggestion
    if (filenameArg.found == polygonStrArg.found) {
```

both shouldn't be provided, right?

Reply:
No, the `filenameArg` is you provide a file containing the polygon, the `polygonStrArg` is if you provide the polygon directly in the command.

Reply:
But if you provide both then we overwrite polygonStr on line 1404, so one of them will be disregarded.

My thought is,
* If both are false (and thus equal), one of them should have been provided, and thus error.
* If both are true (and thus equal), one of them will be disregarded, and thus error.  This is the case my change would address.
* If they have different found values, one of them was found and one of them was not found, so we have exactly one input specifying the polygon.

Reply:
Ahh, I got it, I agree. Just a sec.